src/__tests__/user-event.js

```diff
@@ -24,49 +24,49 @@ test('User events in a form', async () => {
   expect(submitButton).toBeDisabled()
 
   const titleInput = getByLabelText(/title of the movie/i)
-  await userEvent.type(titleInput, fakeReview.title)
+  userEvent.type(titleInput, fakeReview.title)
   expect(titleInput.value).toEqual(fakeReview.title)
 
   const textArea = getByLabelText(/Your review/i)
-  await userEvent.type(textArea, 'The t-rex went insane!')
+  userEvent.type(textArea, 'The t-rex went insane!')
   expect(textArea.value).toEqual('The t-rex went insane!')
 
-  await userEvent.clear(textArea)
+  userEvent.clear(textArea)
   expect(textArea.value).toEqual('')
-  await userEvent.type(textArea, fakeReview.review)
+  userEvent.type(textArea, fakeReview.review)
   expect(textArea.value).toEqual(fakeReview.review)
 
   const initialSelectedRating = getByLabelText(/Awful/i)
   const wonderfulRadioInput = getByLabelText(/Wonderful/i)
   expect(initialSelectedRating).toBeChecked()
   expect(wonderfulRadioInput).not.toBeChecked()
 
-  await userEvent.click(wonderfulRadioInput)
+  userEvent.click(wonderfulRadioInput)
   expect(wonderfulRadioInput).toBeChecked()
-  expect(initialSelectedRating).not.toBeChecked()
+  await waitFor(() => expect(initialSelectedRating).not.toBeChecked())
```

Comment:
Not 100% sure it's better for users to learn when/why they should use `waitFor` instead of just awaiting userSelect? (I honestly don't know 😂)

Reply:
I assume you mean userEvent, not userSelect. I don't think we should encourage users to await `click` because it's not async, and awaiting it implies that the test code will wait for Vue to render (not necessarily true as far as I understand, because Vue renders asyncronously). Awaiting anything before this line (even `await null`) will pass the test. So while `await userEvent.click(...)` is simpler, I don't think it's very correct.

Reply:
👍

Reply:
deal 😉

Reply:
Want to merge this as is, or do you prefer any of the alternatives?

Reply:
I guess it's fine as is 😃 I wonder if ppl will understand/know the difference between "test is failing because Vue hasn't updated the DOM yet" and "test is failing because my component is actually wrong", but that's something beyond our control (as long as Vue updates async)

Reply:
If we make userEvent async for Vue users, this can be reverted, and "test is failing because Vue hasn't updated the DOM yet" should be more obvious when an `await` is missing (some linters can even warn you about not awaiting an async function).